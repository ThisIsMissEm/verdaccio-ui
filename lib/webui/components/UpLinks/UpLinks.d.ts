/**
 * @prettier
 */
import React from 'react';
declare class UpLinks extends React.PureComponent<any> {
    render(): JSX.Element;
    renderUpLinksList: (uplinks: any) => JSX.Element;
    renderContent(uplinks: any, { name }: {
        name: any;
    }): JSX.Element;
}
export default UpLinks;
