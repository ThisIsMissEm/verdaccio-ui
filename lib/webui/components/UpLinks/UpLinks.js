"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @prettier
 */
const react_1 = __importDefault(require("react"));
const List_1 = __importDefault(require("@material-ui/core/List"));
const ListItem_1 = __importDefault(require("@material-ui/core/ListItem"));
const Version_1 = require("../../pages/version/Version");
const NoItems_1 = __importDefault(require("../NoItems"));
const package_1 = require("../../utils/package");
const styles_1 = require("./styles");
class UpLinks extends react_1.default.PureComponent {
    constructor() {
        super(...arguments);
        this.renderUpLinksList = uplinks => (react_1.default.createElement(List_1.default, null, Object.keys(uplinks)
            .reverse()
            .map(name => (react_1.default.createElement(ListItem_1.default, { key: name },
            react_1.default.createElement(styles_1.ListItemText, null, name),
            react_1.default.createElement(styles_1.Spacer, null),
            react_1.default.createElement(styles_1.ListItemText, null, `${package_1.formatDateDistance(uplinks[name].fetched)} ago`))))));
    }
    render() {
        return (react_1.default.createElement(Version_1.DetailContextConsumer, null, ({ packageMeta }) => {
            return this.renderContent(packageMeta._uplinks, packageMeta.latest);
        }));
    }
    renderContent(uplinks, { name }) {
        if (Object.keys(uplinks).length > 0) {
            return (uplinks && (react_1.default.createElement(react_1.default.Fragment, null,
                react_1.default.createElement(styles_1.Heading, { variant: "subheading" }, "Uplinks"),
                this.renderUpLinksList(uplinks))));
        }
        return react_1.default.createElement(NoItems_1.default, { text: `${name} has no uplinks.` });
    }
}
exports.default = UpLinks;
