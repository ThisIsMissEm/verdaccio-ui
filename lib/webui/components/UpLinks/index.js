"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var UpLinks_1 = require("./UpLinks");
exports.default = UpLinks_1.default;
