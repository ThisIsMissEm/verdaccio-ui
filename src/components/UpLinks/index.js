/**
 * @prettier
 */

import { DetailContextConsumer } from '../../pages/version/index';
import { formatDateDistance } from '../../utils/package';
import { Heading, Spacer, ListItemText } from './styles';
import List from '@material-ui/core/List/index';
import ListItem from '@material-ui/core/ListItem/index';
import React from 'react';

class UpLinks extends React.PureComponent<any> {
  render() {
    return (
      <DetailContextConsumer>
        {({ packageMeta }) => {
          return this.renderContent(packageMeta._uplinks);
        }}
      </DetailContextConsumer>
    );
  }

  renderUpLinksList = uplinks => (
    <List>
      {Object.keys(uplinks)
        .reverse()
        .map(name => (
          <ListItem key={name}>
            <ListItemText>{name}</ListItemText>
            <Spacer />
            <ListItemText>{`${formatDateDistance(uplinks[name].fetched)} ago`}</ListItemText>
          </ListItem>
        ))}
    </List>
  );

  renderContent(uplinks) {
    return (
      uplinks && (
        <>
          <Heading variant={'subheading'}>{'Uplinks'}</Heading>
          {this.renderUpLinksList(uplinks)}
        </>
      )
    );
  }
}

export default UpLinks;
