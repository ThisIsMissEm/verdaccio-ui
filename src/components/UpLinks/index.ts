/**
 * @prettier
 */

export { default } from './UpLinks';
