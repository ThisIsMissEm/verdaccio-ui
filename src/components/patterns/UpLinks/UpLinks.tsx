/**
 * @prettier
 */
import React from 'react';
import styled from '@emotion/styled';
import Typography from '@material-ui/core/Typography';
import ListItemText from '@material-ui/core/ListItemText';

import { DetailContext } from '../../../pages/Version';
import List from '@material-ui/core/List/index';
import ListItem from '@material-ui/core/ListItem/index';

const Spacer = styled('div')`
  flex: 1 1 auto;
  border-bottom: 1px dotted rgba(0, 0, 0, 0.2);
  white-space: nowrap;
  height: 0.5em;
`;

interface Props {
  packageMeta: {
    uplinks: any;
  };
}

const UpLinks: React.FC = () => {
  const renderUpLinksList: React.FC<Props> = uplinks => (
    <List>
      {Object.keys(uplinks)
        .reverse()
        .map(name => (
          <ListItem key={name}>
            <ListItemText>{name}</ListItemText>
            <Spacer />
            {/* <ListItemText>{`${formatDateDistance(uplinks[name].fetched)} ago`}</ListItemText> */}
          </ListItem>
        ))}
    </List>
  );

  const renderContent: React.FC<Props> = ({ packageMeta: { uplinks } }) =>
    uplinks && (
      <>
        <Typography variant={'subheading'}>{'Uplinks'}</Typography>
        {renderUpLinksList(uplinks)}
      </>
    );

  return <DetailContext.Consumer>{context => context && renderContent(context)}</DetailContext.Consumer>;
};

export default UpLinks;
