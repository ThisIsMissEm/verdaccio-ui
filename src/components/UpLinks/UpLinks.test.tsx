import React from 'react';
import { shallow } from 'enzyme';

import UpLinks from './UpLinks';

describe('<UpLinks /> component', () => {
  test('should render the component in default state', () => {
    const wrapper = shallow(<UpLinks />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
