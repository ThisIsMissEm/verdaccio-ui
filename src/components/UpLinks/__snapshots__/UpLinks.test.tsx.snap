// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpLinks /> component should render the component in default state 1`] = `""`;
